src/controllers/sdex/index.ts

```diff
@@ -72,4 +83,95 @@ router.get(
   }),
 );
 
+router.get(
+  '/swaps',
+  asyncRoute(async (req, res) => {
+    const { user, chainId } = validate(swapHistoryQuerySchema, req.query);
+    const p = validatePaginatedRequest(req);
+    const offset = (p.page - 1) * p.limit;
+    const cacheKey = `sdex/swaps/${chainId}/${user}/${p.limit}/${p.cursor}`;
+
+    return maybeCacheResponse(
+      res,
+      cacheKey,
+      async () => {
+        const swapsQuery = db
+          .select({
+            chainId: swapsTableV2.chainId,
+            transactionHash: swapsTableV2.transactionHash,
+            baseAmount: swapsTableV2.baseAmount,
+            quoteAmount: swapsTableV2.quoteAmount,
+            fees: swapsTableV2.fees,
+            callIndex: swapsTableV2.callIndex,
+            baseId: swapsTableV2.baseId,
+            quoteId: swapsTableV2.quoteId,
+            user: swapsTableV2.user,
+            block: swapsTableV2.block,
+            tickAt: swapsTableV2.tickAt,
+          })
+          .from(swapsTableV2)
+          .where(
+            and(
+              chainId ? eq(swapsTableV2.chainId, chainId) : undefined,
+              user ? eq(swapsTableV2.user, user) : undefined,
+            ),
+          )
+          .limit(p.limit)
+          .offset(offset)
+          .$dynamic();
+
+        const api = createApiQuery('transactionHash', OrderBy.asc, (key) => swapsTableV2[key], p);
```

Comment:
newest transactions should be shown first in the list

```suggestion
        const api = createApiQuery('id', OrderBy.desc, (key) => swapsTableV2[key], p);
```